### ISSUE TITLE
CGAL-4.13 configure failed with VTK master (VTK 9)

### ISSUE THREAD

**Issue by beew** [CGAL/cgal#3789]
OS is ubuntu 16.04
cmake version 3.11.4
gcc -5.4.0

Tried to configure cgal-4.13 with
```
cmake -DCMAKE_BUILD_TYPE=Release -DWITH_examples=true -DWITH_demos=true -DWITH_BLAS=ON -DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda -DCUDA_SDK_ROOT_DIR=$HOME/opt/cuda-samples/common -DBUILD_SHARED_LIBS=ON -DBUILD_TESTING=ON -DWITH_CGAL_ImageIO=ON -DWITH_CGAL_Qt5=ON -DWITH_GMP=ON -DWITH_LAPACK=ON -DWITH_MPFI=ON -DWITH_MPFR=ON -DWITH_NTL=ON -DWITH_OpenGL=ON -DWITH_ZLIB=ON -DWITH_QGLViewer=ON -DWITH_CGAL_Core=ON -DQGLVIEWER_INCLUDE_DIR=/home/bernard/opt/libQGLViewer/include -DQGLVIEWER_LIBRARY_RELEASE=/home/bernard/opt/libQGLViewer/lib/libQGLViewer-qt5.so -DCIMG_INCLUDE_DIR=/home/bernard/opt/cimg -DWITH_tests=ON -DSuiteSparse_UMF_INCLUDE_DIR=$HOME/opt/suitesparse/include -DSuiteSparse_UMFPACK_LIBRARY_RELEASE=$HOME/opt/suitesparse/lib/libumfpack.so ..
```
and got these errors
```
-- Found OpenCV 4.0.1
CMake Error at Polyhedron/demo/Polyhedron/Plugins/IO/CMakeLists.txt:42 (include):
  include called with wrong number of arguments.  include() only takes one
  file.
...
-- Configuring demo in demo/Surface_mesher
-- Build type: Release
-- USING CXXFLAGS = ' -O3 -DNDEBUG'
-- USING LDFLAGS = ' '
-- Requested component: ImageIO
-- Requested component: Qt5
-- Requested component: MPFI
-- UseMPFI
-- MPFI include:      /usr/local/include
-- MPFI definitions:  
-- MPFI libraries:    /usr/local/lib/libmpfi.so
-- Configured MPFI from UseLIB-file: CGAL_UseMPFI
-- Requested component: MPFR
-- Requested component: GMP
-- Requested component: NTL
CMake Error at Surface_mesher/demo/Surface_mesher/CMakeLists.txt:100 (include):
  include called with wrong number of arguments.  include() only takes one
  file.
```

The error log
```
Determining if the pthread_create exist failed with the following output:
Change Dir: /home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_11b21/fast"
/usr/bin/make -f CMakeFiles/cmTC_11b21.dir/build.make CMakeFiles/cmTC_11b21.dir/build
make[1]: Entering directory '/home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_11b21.dir/CheckSymbolExists.c.o
/usr/bin/cc -DLAPACK_USE_F2C   -o CMakeFiles/cmTC_11b21.dir/CheckSymbolExists.c.o   -c /home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_11b21
/home/bernard/opt/cmake/bin/cmake -E cmake_link_script CMakeFiles/cmTC_11b21.dir/link.txt --verbose=1
/usr/bin/cc      -rdynamic CMakeFiles/cmTC_11b21.dir/CheckSymbolExists.c.o  -o cmTC_11b21 
CMakeFiles/cmTC_11b21.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x16): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_11b21.dir/build.make:86: recipe for target 'cmTC_11b21' failed
make[1]: *** [cmTC_11b21] Error 1
make[1]: Leaving directory '/home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_11b21/fast' failed
make: *** [cmTC_11b21/fast] Error 2

File /home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_2ebe3/fast"
/usr/bin/make -f CMakeFiles/cmTC_2ebe3.dir/build.make CMakeFiles/cmTC_2ebe3.dir/build
make[1]: Entering directory '/home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_2ebe3.dir/CheckFunctionExists.c.o
/usr/bin/cc -DLAPACK_USE_F2C  -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTC_2ebe3.dir/CheckFunctionExists.c.o   -c /home/bernard/opt/cmake/share/cmake-3.11/Modules/CheckFunctionExists.c
Linking C executable cmTC_2ebe3
/home/bernard/opt/cmake/bin/cmake -E cmake_link_script CMakeFiles/cmTC_2ebe3.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create    -rdynamic CMakeFiles/cmTC_2ebe3.dir/CheckFunctionExists.c.o  -o cmTC_2ebe3 -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_2ebe3.dir/build.make:86: recipe for target 'cmTC_2ebe3' failed
make[1]: *** [cmTC_2ebe3] Error 1
make[1]: Leaving directory '/home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_2ebe3/fast' failed
make: *** [cmTC_2ebe3/fast] Error 2


Performing C++ SOURCE FILE Test TBB_without_pthread failed with the following output:
Change Dir: /home/bernard/opt/cgal-releases-CGAL-4.13/build/CMakeFiles/CMakeTmp
```

Thanks.

Edited: same error with opencv-3.4.5

**Comment by beew** [comment]
But removing the folders Polyhedron, Surface_mesher compiling went through. The culprit is this line
`include(${VTK_USE_FILE})`
in Polyhedron/demo/Polyhedron/Plugins/IO/CMakeLists and Surface_mesher/demo/Surface_mesher/CMakeLists
How to edit them to make it work?

**Comment by lrineau** [comment]
Which version of VTK did you use? It seems CMake set `VTK_FOUND` to `TRUE`, but `VTK_USE_FILE` is empty. Maybe you are using the `master` version of VTK, targeting the future VTK version 9. Kitware has changed a lot in the CMake scripts of VTK, and it seems that `VTK_USE_FILE` no longer exists.

If you want to workaround that issue, you can set the CMake variable `CMAKE_DISABLE_FIND_PACKAGE_VTK` to `TRUE`.

**Comment by beew** [comment]
Yes VTK master

**Comment by lrineau** [comment]
For the moment (CGAL-4.13):
  - either set the CMake variable `CMAKE_DISABLE_FIND_PACKAGE_VTK` to `TRUE`, to disable the VTK support in CGAL,
  - or use VTK < 9.

For CGAL-4.14, I will try to support the new VTK version.

**Comment by lrineau** [comment]
This issue is now fixed by the merge of PR #4027, but only in the branch for CGAL-4.14.1 and in the `master` branch (for CGAL-5.0).
